index.d.ts

```diff
@@ -0,0 +1,78 @@
+import { Action, StoreEnhancer, Store } from 'redux';
+
+interface StoreCreator {
+  <S, T>(
+    reducer: EnhancedReducer<S, T>,
+    preloadedState: S,
+    enhancer: StoreEnhancer<S>
+  ): Store<S>;
+}
+
+interface EnhancedReducer<S, T> {
+  <A extends Action>(state: S, action: A): T;
+}
+
+interface NoneCmd {
+  type: 'NONE';
+}
+
+interface ActionCmd<A> {
+  type: 'ACTION';
+  actionToDispatch: A;
+}
+
+interface BatchCmd<AppAction extends Action, F extends Function> {
+  type: 'BATCH';
+  cmds: CmdType<AppAction, F>;
```

Comment:
should this be `CmdType<AppAction, F>[]` since it's an array of them?

Reply:
👍  not sure how these slipped by 😓